sdk/schemaregistry/Microsoft.Azure.Data.SchemaRegistry.ApacheAvro/tests/SchemaRegistryAvroObjectSerializerLiveTest.cs

```diff
@@ -0,0 +1,96 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using Avro;
+using Avro.Generic;
+using Azure.Core.TestFramework;
+using Azure.Data.SchemaRegistry;
+using NUnit.Framework;
+using System;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+using TestSchema;
+
+namespace Microsoft.Azure.Data.SchemaRegistry.ApacheAvro.Tests
+{
+    public class SchemaRegistryAvroObjectSerializerLiveTest : RecordedTestBase<SchemaRegistryClientTestEnvironment>
+    {
+        public SchemaRegistryAvroObjectSerializerLiveTest(bool isAsync) : base(isAsync)
+        {
+            TestDiagnostics = false;
+        }
+
+        private SchemaRegistryClient CreateClient() =>
+            InstrumentClient(new SchemaRegistryClient(
+                TestEnvironment.SchemaRegistryUri,
+                TestEnvironment.Credential,
+                Recording.InstrumentClientOptions(new SchemaRegistryClientOptions())
+            ));
+
+        [Test]
+        public async Task CanSerializeAndDeserialize()
+        {
+            var client = CreateClient();
+            var groupName = "miyanni_srgroup";
+            var employee = new Employee { Age = 42, Name = "Caketown" };
+
+            var memoryStream = new MemoryStream();
+            var serializer = new SchemaRegistryAvroObjectSerializer(client, groupName, new SchemaRegistryAvroObjectSerializerOptions { AutoRegisterSchemas = true });
+            await serializer.SerializeAsync(memoryStream, employee, typeof(Employee), CancellationToken.None);
+
+            memoryStream.Position = 0;
+            var deserializedObject = await serializer.DeserializeAsync(memoryStream, typeof(Employee), CancellationToken.None);
+            var readEmployee = deserializedObject as Employee;
+            Assert.IsNotNull(readEmployee);
+            Assert.AreEqual("Caketown", readEmployee.Name);
+            Assert.AreEqual(42, readEmployee.Age);
+        }
+
+        [Test]
+        public async Task CanSerializeAndDeserializeGenericRecord()
+        {
+            var client = CreateClient();
+            var groupName = "miyanni_srgroup";
+            var record = new GenericRecord((RecordSchema)Employee._SCHEMA);
+            record.Add("Name", "Caketown");
+            record.Add("Age", 42);
+
+            var memoryStream = new MemoryStream();
```

Comment:
nit:  `MemoryStream` is disposable.

Reply:
Good find! I used `using` in my code correctly but forgot it on the tests and snippets. Fixed.